src/java/org/apache/cassandra/net/Message.java

```diff
@@ -473,6 +482,13 @@ public Builder<T> withParam(ParamType type, Object value)
             return this;
         }
 
+        public Builder<T> withCustomParam(String name, byte[] value)
```

Comment:
May want to provide a method that takes a prebuilt map in its entirety for those users of the API who would prefer that to doing some unnecessary look-ups per param.